### ISSUE TITLE
Belt printer support on OrcaSlicer

### ISSUE THREAD

**Issue by vikichu** [SoftFever/OrcaSlicer#6885]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

No, this is an additional feature.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

There is an existing issue: Belt support #2628, but this issues isn't labeled as "Feature Request".
 
Slicer should have functionality to support belt printer of varying degrees. Typical is 45 degrees. GUI should render out sliced objects properly.

### Describe alternatives you've considered

1. Ideamaker as slicer: Useable, but not as good of an experience as OrcaSlicer/ PrusaSlicer derivatives.
2. Cura: Haven't tried this slicer for belt printer.

**Comment by enzoav** [comment]
+1

As well as all of the +1's in this thread: https://github.com/SoftFever/OrcaSlicer/issues/2628

🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾🙏🏾

**Comment by emanuele1412** [comment]
+1 here
Belt printers are getting more popular and with the klipper enabled IR3 V2 coming to the market, there will be a big demand for a good slicer that is capable to print at 45 degree angle.
Currently only Ideamaker slicer can do this but is extremely basic, slow, overcomplicated and featureless. An advanced slicer like Orca should could be lightyears ahead if it gets develepoment towards belt printing and would revolutionize the marlet of such software.

**Comment by Jerichoulr** [comment]
+1 what do we have to do to get this implemented is there a way to "donate" to get this feature added? I would love to use orca slicer with the idea former ir 3 v2

**Comment by enzoav** [comment]
> +1 what do we have to do to get this implemented is there a way to "donate" to get this feature added? I would love to use orca slicer with the idea former ir 3 v2

I too will donate whatever is needed to get this ball rolling.  Belt printing needs Orca Slicer to save it. 🙏🏾🙏🏾🙏🏾

**Comment by d-portero** [comment]
+1

**Comment by D3printingFL** [comment]
What do we have to do to get this more attention?

**Comment by enzoav** [comment]
🤷🏾‍♂️

**Comment by xboxhacker** [comment]
+1 
+1 
+1 
I have 3 belt printers.

**Comment by enzoav** [comment]
> What do we have to do to get this more attention?

start throwing money at someone...

**Comment by Jerichoulr** [comment]
Who do we have to throw money at?

**Comment by basz** [comment]
I think it's not *that* complicated. before slicing a matrix transformation needs to be applied (taking the angle into consideration). I have played with this to figure out how it could work in threejs.

Then, when viewing the sliced gcode the viewer should apply the reverse of the original matrix before rendering so that the geometry is back into its original form.

First layers modifications should be applied in a post processing of the gcode step of the gcode. (as on a belt printer the first layer is actually accross many layers.

I'm sure I'm missing some things but that should be the theory...

**Comment by Justjohn1982** [comment]
+1 for this feature!

**Comment by bathtub-toast** [comment]
+1 from me as well, somewhat motivated by the IdeaFormer IR3 v2 running a test print behind me. Also motivated by how much better an experience Orca has been for me compared to Cura, I refuse to go back.

**Comment by martinjohuber** [comment]
+1 would love to see beltprinter support, if you need any support I can help you

**Comment by grzbiet** [comment]
+1 and a question - what do you recommend for slicing while we wait for orca infinite?

**Comment by Alaric323** [comment]
> +1 and a question - what do you recommend for slicing while we wait for orca infinite?

For now, the only major slicer that has belt support built in appears to be IdeaMaker.  Most other implementations are bodged together versions of Cura, like Blackbelt Cura and Creality Slicer.

**Comment by actinaria** [comment]
Definitely would love to bee belt printers integrated into Orca

**Comment by KazeOne** [comment]
I too would like to see such support, would even donate towards it's development

**Comment by jessegra** [comment]
+1 +$$$

**Comment by d-portero** [comment]
Someone  posted a GoFundMe to try and pool money to get this worked on https://www.gofundme.com/f/get-3d-belt-printers-into-orca-slicer

https://github.com/SoftFever/OrcaSlicer/issues/2628

**Comment by xboxhacker** [comment]
If anyone wants to give it a try.....

I have revived an 8 year old script to allow you to slice on Orca and print on a belt printer. It can be found on my repo.

[link](https://github.com/xboxhacker/Tilted-Bed-Conveyor)

Download the release or build your own from the C# code.

This is currently working for me on my IR3V2. It _should_ work for the V1 and CR30 as well.

**Comment by Leorizer** [comment]
+1 I have a IdeaFormer IR3 V2 and I realy ´dislike the experience of the IdeaFormer Slicer (It Succs) the only OK way of slicing for Belt printers i found is a Cura Belt plugin but it is on a older Cura version and Supports/Brim and other stuff doesn't work at all.
Id love to use Orca or Prusaslicer for the IR3 V2

**Comment by tomfc262** [comment]
YES Please I use orca for 7 bambu machines and just bought a ideaformer-3d ir3 v2 for repeating prints want to keep everything under one slicer that just plain works

**Comment by nikitax75** [comment]
+1 Have a printfarm with 20 printers and among them 4 IR3V2, probably one the best core XY we had, with the belt it's a killer printer. Just works well, fast and is reliable. Orca needs to support this printer, they're selling them by the hundred

**Comment by hegge25** [comment]
+1. Yes please, would love to use orca for my ideaformer IR3V2

**Comment by csscsccs** [comment]
> If anyone wants to give it a try.....
> 
> I have revived an 8 year old script to allow you to slice on Orca and print on a belt printer. It can be found on my repo.
> 
> https://github.com/xboxhacker/Tilted-Bed-Conveyor
> 
> Download the release or build your own from the C# code.
> 
> This is currently working for me on my IR3V2. It _should_ work for the V1 and CR30 as well.
> 
> **This is just a band-aid!!** _You can not use slicer supports with this._ You can model them into your STL tho.
> 
> ## We still need 100% native Orca slicing.

when i use it's gcode,I always got this error"!! Move out of range: 97.203 -5.784 0.283 [75.697]"

**Comment by neusilon** [comment]
Would using conical slicing, as seen in this CNC Kitchen video, be a good substitute until belt support is implemented?
https://www.youtube.com/watch?v=1i-1TEdByZY

Update: It kinda works, maybe for small parts. I'm trying to print a 31" rotor blade, and there is some warping with the conical slicer, and you can't use supports with this method. I also used @xboxhacker method, which is much quicker to use. Hopefully, someone with more experience can make this work.

**Comment by xboxhacker** [comment]
> when i use it's gcode,I always got this error"!! Move out of range: 97.203 -5.784 0.283 [75.697]"

Change this in your `printer.cfg`

```
[stepper_y]
position_min: -6.0
```

This will allow for a lower Y movement on the belt. Make sure you do not crash the nozzle! Keep an eye on it.

**Comment by petros-tinkers** [comment]
I would love to get belt support for orca. Orca is the best slicer ever.

Dear developers, If you decide to make this happen, make the angle configurable including negative degrees. I am building an experimental printer and tried using Ideamaker for testing but it wouldn't support my printers geometry due to this (it only accepts positive angle values).

**Comment by marcusg101** [comment]
+1 for this! I have 2 home brew belt printers. Ideamaker is good but Orca is so much better!

**Comment by kevinahuber** [comment]
@SoftFever is there any organizational or philosophical why this hasn't been prioritized, or is it quite simply a hard feature and needs contributors? 

Seeing so much enthusiasm between this thread and #2628

**Comment by bathtubtoast** [comment]
@kevinahuber when I dug into why, it appears that slic3r is the base reason as it does not support an angled slicing start, but as I'm not sure which version Orca pulls from I have not just blindly put any requests out there.

**Comment by fishienoob** [comment]
+1

**Comment by cupofRiesL** [comment]
+1 I have a CR-30 Printmill and would love to have a newer and better slicer

**Comment by Joebus1** [comment]
+1 CR-30 User. Current OrcaSlicer user.

**Comment by marcusg101** [comment]
Yeah I'm using a custom ender 3 that I modded with a kit. It needs proper
software.

**Comment by dandv77** [comment]
I tried opening a new enhancement request but it was directed here.  Why can't the CR-30 info in Creality Print, which is a fork of Orca therefore should be opensource, be imported and used for a template for belt printers?
